### PR TITLE
Appstore connect feature branch

### PR DESCRIPTION
**Pull request by flub**
Now that PyJWT upgrade is merged and deployed this should be able to get merged into master.

### CODE REVIEW

**Comment by flub** [comment]
@priscilawebdev could you also check out the visual snapshot change?  looks weird to me as even the font seems to have changed?

File: src/sentry/api/endpoints/project_app_store_connect_credentials.py [hunk 1]
```diff
@@ -380,15 +546,34 @@ def post(self, request, project):
 
 
 class AppStoreConnectRequestSmsSerializer(serializers.Serializer):
+    """Input validation for :class:`AppStoreConnectRequestSmsEndpoint`."""
+
     sessionContext = serializers.CharField(min_length=1, required=True)
 
 
 class AppStoreConnectRequestSmsEndpoint(ProjectEndpoint):
+    """Switches an ITunes login to using SMS for 2FA.
```
Comment (priscilawebdev) [inline src/sentry/api/endpoints/project_app_store_connect_credentials.py:555]:
```suggestion
    """Switches an iTunes login to using SMS for 2FA.
```

File: src/sentry/api/endpoints/project_app_store_connect_credentials.py [hunk 2]
```diff
@@ -380,15 +546,34 @@ def post(self, request, project):
...
+    ```json
+    {
+        "sessionContext": "xxxx"
+    }
+    ```
+
+    The response will contain a new ``sessionContext`` which must be used in the call to
+    :class:`AppStoreConnect2FactorAuthEndpoint`
+    (``projects/{org_slug}/{proj_slug}/appstoreconnect/2fa/``) to complete the login of this
+    ITunes session.
```
Comment (priscilawebdev) [inline src/sentry/api/endpoints/project_app_store_connect_credentials.py:569]:
```suggestion
    iTunes session.
```

File: src/sentry/api/endpoints/project_app_store_connect_credentials.py [hunk 3]
```diff
@@ -1,3 +1,54 @@
+"""Sentry API to manage the App Store Connect credentials for a project.
+
+There are currently two sets of credentials required:
+- API credentials
+- ITunes credentials
```
Comment (priscilawebdev) [inline src/sentry/api/endpoints/project_app_store_connect_credentials.py:5]:
```suggestion
- iTunes credentials
```

File: src/sentry/api/endpoints/project_app_store_connect_credentials.py [hunk 4]
```diff
@@ -1,3 +1,54 @@
...
+To create and manage these credentials, several API endpoints exist:
+
+1. ``POST projects/{org_slug}/{proj_slug}/appstoreconnect/apps/``
+
+   This will validate the API credentials and return the list of available applications if
+   valid, or 401 if invalid.  See :class:`AppStoreConnectAppsEndpoint`.
+
+2. ``POST projects/{org_slug}/{proj_slug}/appstoreconnect/start/``
+
+   This starts an ITunes login, either brand new credentials or re-authenticate an existing
```
Comment (priscilawebdev) [inline src/sentry/api/endpoints/project_app_store_connect_credentials.py:20]:
```suggestion
   This starts an iTunes login, either brand new credentials or re-authenticate an existing
```

File: src/sentry/api/endpoints/project_app_store_connect_credentials.py [hunk 5]
```diff
@@ -444,17 +629,59 @@ def post(self, request, project):
...
 class AppStoreConnect2FactorAuthSerializer(serializers.Serializer):
+    """Input validation for :class:`AppStoreConnect2FactorAuthEndpoint."""
+
     code = serializers.CharField(max_length=10, required=True)
     useSms = serializers.BooleanField(required=True)
     sessionContext = serializers.CharField(min_length=1, required=True)
 
 
 class AppStoreConnect2FactorAuthEndpoint(ProjectEndpoint):
+    """Completes the 2FA ITunes login, returning a valid session.
```
Comment (priscilawebdev) [inline src/sentry/api/endpoints/project_app_store_connect_credentials.py:640]:
```suggestion
    """Completes the 2FA iTunes login, returning a valid session.
```

File: src/sentry/api/endpoints/project_app_store_connect_credentials.py [hunk 6]
```diff
@@ -181,11 +286,23 @@ class AppStoreUpdateCredentialsSerializer(serializers.Serializer):
...
 class AppStoreConnectUpdateCredentialsEndpoint(ProjectEndpoint):
+    """Updates a subset of the existing credentials.
+
+    ``POST projects/{org_slug}/{proj_slug}/appstoreconnect/{id}/``
+
+    See :class:`AppStoreUpdateCredentialsSerializer` for the input format.
+
+    This is like :class:`AppStoreConnectCreateCredentialsEndpoint` aka
+    ``projects/{org_slug}/{proj_slug}/appstoreconnect/{id}/`` but allows you to only provide
+    a sub-set.  This is most useful when you had to refresh the ITunes session using
```
Comment (priscilawebdev) [inline src/sentry/api/endpoints/project_app_store_connect_credentials.py:297]:
```suggestion
    a sub-set.  This is most useful when you had to refresh the iTunes session using
```

**Comment by priscilawebdev** [comment]
@flub the visual change seems unrelated 😉

**Review by priscilawebdev — APPROVED** [review]

**Pull request merged**